hazelcast-client-new/src/test/java/com/hazelcast/client/ringbuffer/RingbufferTest.java

```diff
@@ -0,0 +1,223 @@
+package com.hazelcast.client.ringbuffer;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.RingbufferConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.core.IFunction;
+import com.hazelcast.ringbuffer.ReadResultSet;
+import com.hazelcast.ringbuffer.Ringbuffer;
+import com.hazelcast.ringbuffer.impl.client.PortableReadResultSet;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.Future;
+
+import static com.hazelcast.ringbuffer.OverflowPolicy.OVERWRITE;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class RingbufferTest extends HazelcastTestSupport {
+
+    public static int CAPACITY = 10;
+
+    static HazelcastInstance client;
+    static HazelcastInstance server;
+    private Ringbuffer<String> clientRingbuffer;
+    private Ringbuffer<String> serverRingbuffer;
+
+    @BeforeClass
+    public static void init() {
+        setLoggingLog4j();
+        Config config = new Config();
+        config.addRingBufferConfig(new RingbufferConfig("rb*").setCapacity(CAPACITY));
+
+        server = Hazelcast.newHazelcastInstance(config);
```

Comment:
Almost all client tests are using it right now. 
Example, https://github.com/hazelcast/hazelcast/blob/master/hazelcast-client-new/src/test/java/com/hazelcast/client/map/ClientMapTest.java